Rx/v2/test/operators/publish.cpp

```diff
@@ -38,6 +40,104 @@ SCENARIO("publish range", "[!hide][range][subject][publish][subject][operators]"
     }
 }
 
+SCENARIO("publish ref_count", "[range][subject][publish][ref_count][operators]"){
```

Comment:
I notice that these tests do not use the test scheduler or the on.next(time, value), etc.. utility. I would like to see these tests follow the normal pattern.

Reply:
I was a bit confused by the functionality of the test scheduler. In this case, I wanted to exercise that the 'same-thread' (i.e. no scheduler) is doing the right thing. Was that the wrong thing to test?

What is the recommended pattern to test that?

Reply:
the test scheduler does all work on the current thread. it is a good way to test this. it also records the marble values and when they happen.

Reply:
I believe I made the test changes requested, see 747366b